### PR TITLE
Provide pytest plugin to simplify testing in sub-projects

### PR DESCRIPTION
**Pull request by rodja**
This PR allows projects based on RoSys to use the here defined fixtures for their own testing. Before it was necessary to copy & paste the fixtures.

### CODE REVIEW

**Comment by rodja** [comment]
A usage demonstration can be seen in the pull request https://github.com/zauberzeug/field_friend/pull/50 in the Field Friend repository.

File: tests/test_approx.py [hunk 1]
```diff
@@ -64,3 +65,4 @@ class A:
     approx(A(x=1), A(x=1))
     with pytest.raises(AssertionError):
         approx(A(x=1), A(x=2))
+        approx(A(x=1), A(x=2))
```
Comment (falkoschindler) [inline tests/test_approx.py:68]: Why did you duplicate this line?

File: test.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import multiprocessing
```
Comment (falkoschindler) [inline test.py:1]: Did you add this file intentionally?

File: tests/test_odometer.py [hunk 3]
```diff
@@ -38,3 +39,4 @@ def test_odometry_frame():
     detection = Pose(x=1.5, y=4.0, yaw=np.deg2rad(135), time=2)
     odometry_frame = Odometer._compute_odometry_frame(local_pose, detection)
     approx(odometry_frame, Pose(x=1.0, y=2.0, yaw=np.deg2rad(90), time=2))
+    approx(odometry_frame, Pose(x=1.0, y=2.0, yaw=np.deg2rad(90), time=2))
```
Comment (falkoschindler) [inline tests/test_odometer.py:42]: Why did you duplicate this line?

File: pyproject.toml [hunk 4]
```diff
@@ -72,4 +72,15 @@ requires = [
...
+
+[tool.pytest.ini_options]
+asyncio_mode = "auto"
+# NOTE: changing default location of pytest_cache because the uvicorn file watcher somehow triggered too many reloads
+cache_dir = "../.pytest_cache" 
+testpaths = [
+    "tests"
+]
+
+[tool.poetry.plugins.pytest11]
```
Comment (falkoschindler) [inline pyproject.toml:85]: What the purpose of this block?

**Review by falkoschindler — COMMENTED** [review]
@rodja I've added some comments about this PR. Maybe we can discuss them tomorrow.

File: tests/test_odometer.py (hunk 3 above)
Comment (rodja) [reply tests/test_odometer.py]: Accident.

File: test.py (hunk 2 above)
Comment (rodja) [reply test.py]: Accident.

File: tests/test_approx.py (hunk 1 above)
Comment (rodja) [reply tests/test_approx.py]: Accident.

File: pyproject.toml (hunk 4 above)
Comment (rodja) [reply pyproject.toml:85]: It indicates the start of settings related to pytest plugin. pytest11 refers to the plugin architecture. By offering a `pytest-rosys` plugin, other sub-projects can use the fixtures from rosys/testing/fixtures.py.

**Review by falkoschindler — APPROVED** [review]

**Pull request merged**